ui/app/models/volume.js

```diff
@@ -1,10 +1,42 @@
+import { computed } from '@ember/object';
+import Model from 'ember-data/model';
 import attr from 'ember-data/attr';
-import Fragment from 'ember-data-model-fragments/fragment';
+import { belongsTo, hasMany } from 'ember-data/relationships';
 
-export default Fragment.extend({
+export default Model.extend({
+  plainId: attr('string'),
   name: attr('string'),
 
-  source: attr('string'),
-  type: attr('string'),
-  readOnly: attr('boolean'),
+  namespace: belongsTo('namespace'),
+  plugin: belongsTo('plugin'),
+
+  writeAllocations: hasMany('allocation'),
+  readAllocations: hasMany('allocation'),
+
+  externalId: attr('string'),
+  topologies: attr(),
+  accessMode: attr('string'),
+  attachmentMode: attr('string'),
+  schedulable: attr('boolean'),
+  provider: attr('string'),
+  version: attr('string'),
+
+  controllerRequired: attr('boolean'),
+  controllersHealthy: attr('number'),
+  controllersExpected: attr('number'),
+
+  controllersHealthyPercent: computed('controllersHealthy', 'controllersExpected', function() {
+    return this.controllersHealthy / this.controllersExpected;
+  }),
+
+  nodesHealthy: attr('number'),
+  nodesExpected: attr('number'),
+
+  nodesHealthyPercent: computed('nodesHealthy', 'nodesExpected', function() {
+    return this.nodesHealthy / this.nodesExpected;
+  }),
```

Comment:
Small point, perhaps nitpicky, but wouldn’t these be proportions rather than percents?

Reply:
They absolutely would. I can change the names. As you probably noticed, they aren't actually printed anywhere–just used as way to sort.